src/database/queries/users/selectWithPagination.sql

```diff
@@ -2,8 +2,8 @@ SELECT user_to_json(users.*) as "object"
 FROM users
 WHERE
   CASE
-    WHEN :authenticationId != '' THEN
-      authentication_id = :authenticationId
+    WHEN :keycloakUserId != '' THEN
+      keycloak_user_id = :keycloakUserId::uuid
```

Comment:
I don't think you need an explicit cast here; the equals operator will implicitly cast as necessary except when it's ambiguous.

Reply:
I added this in response to an error I was getting from psql complaining that there was no such thing as a conversion from text to uuid.

I'll try removing it and get back to you, maybe this solution was a red herring.

Reply:
Oh, that sounds plausible - maybe it's different in a prepared transaction than in `psql`. If so, nevermind, and sorry for the hassle!

Reply:
Yeah -- I did try taking this out in the context of a NULL check and it didn't like me one bit (claimed to not be able to tell what the type is).

I think this has to be there :( marking as resolved for now.